packages/contracts/contracts/SmartEscrow.sol

```diff
@@ -187,7 +186,7 @@ contract SmartEscrow is Context, IArbitrable, ReentrancyGuard {
   function payDisputeFees(address _adr) internal {
     IArbitrator arbitrator = IArbitrator(_adr);
     (, IERC20 feeToken, uint256 feeAmount) = arbitrator.getDisputeFees();
-    feeToken.safeTransferFrom(msg.sender, address(this), feeAmount);
+    feeToken.safeTransferFrom(msg.sender, address(this), feeAmount); // sender must pay dispute fees when locking
```

Comment:
> line 206: how are we determining how to pay the dispute fees?

in factory version, we can have a different approach where we can check if the contract has enough balance for the fees, then sender may not need to pay the dispute fees.

```suggestion
    uint256 balance = feeToken.balanceOf(address(this));
    if (balance < feeAmount) {
        feeToken.safeTransferFrom(msg.sender, address(this), feeAmount.sub(balance); // sender must pay dispute fees when locking
    }
```

BUT while implementing this I thought that it makes sense for the user who's raising the dispute to pay the fees. 
I'm not sure what's the right approach here. Open to suggestions.